src/wp-includes/class-wp-object-cache.php

```diff
@@ -542,6 +570,34 @@ public function reset() {
 		}
 	}
 
+	/**
+	 * Serves as a utility function to determine whether a key is valid.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @param int|string $key   Cache key to check for validity.
+	 * @return bool Whether the key is valid.
+	 */
+	protected function is_valid_key( $key ) {
+		if ( is_int( $key ) ) {
+			return true;
+		}
+
+		if ( is_string( $key ) && trim( $key ) !== '' ) {
+			return true;
+		}
+
+		$type = str_replace( 'string', 'empty string', gettype( $key ) );
+
+		_doing_it_wrong(
+			debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS )[1]['function'],
```

Comment:
Could we check to see if this key exists before using it?

Reply:
I'm happy to add a check, but it's not needed since the method is protected and the there will always be a caller.

What do you suggest as a fallback if `[1]['function']` doesn't exist?

Reply:
You could pass the ``__FUNCTION__`` as second param to the function.

Reply:
Could you please give an example where this key does not exist?
Bc afaik this is technically impossible on protected methods

Reply:
I've tried to find cases where `[1]` wouldn't exist when the method is protected and can't find any.

```php

class cache {
  public function add() {
    $this->is_valid_key();

    $foo = function() {
      $this->is_valid_key();
    };
    
    $foo();

    call_user_func([$this, 'is_valid_key']);
  }

  protected function is_valid_key() {
    var_dump(
      debug_backtrace( DEBUG_BACKTRACE_IGNORE_ARGS )[1]['function']
    );
  }
}

(new cache)->add();
```

Reply:
However, we can use `wp_debug_backtrace_summary()` instead and append either it's string to the error, or use it to fetch the last caller?

Reply:
Did you performance measure wp_debug_backtrace_summary against what you have now? Bc I think its's much slower